tsdb/head.go

```diff
@@ -1068,13 +1069,14 @@ func (a *headAppender) Commit() error {
 
 	a.head.metrics.samplesAppended.Add(float64(total))
 	a.head.updateMinMaxTime(a.mint, a.maxt)
-	a.head.iso.closeAppend(a.appendID)
 
 	return nil
 }
 
 func (a *headAppender) Rollback() error {
-	a.head.metrics.activeAppenders.Dec()
+	defer a.head.metrics.activeAppenders.Dec()
```

Comment:
Why defer rather than immediately?

Reply:
To be consistent with what we do in `Commit`. I also think it is clearer to do it in the end in this way. Makes debugging easier if for example one of the locks cannot get acquired.